.circleci/config.yml

```diff
@@ -1,76 +1,58 @@
 # something different.  Probably best to avoid that, out of principle, though.
-common_test_steps: &common_test_steps
-  steps:
-    - *run_install_desired_npm
-    - checkout
-    - restore_cache:
-        keys:
-          # When lock file changes, use increasingly general patterns to restore cache
-          - npm-v2-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "package-lock.json" }}
-          - npm-v2-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-
-          - npm-v2-{{ .Environment.CACHE_VERSION }}-
-    - run: npm --version
-    - run: npm ci
-    - save_cache:
-        key: npm-v2-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "package-lock.json" }}
-        paths:
-          # This should cache the npm cache instead of node_modules, which is needed because
-          # npm ci actually removes node_modules before installing to guarantee a clean slate.
-          - ~/.npm
-    - run: npm run circle
+  common_test_steps:
+    description: Commands to run on every Node.js environment
+    steps:
+      - oss/install_specific_npm_version
+      - checkout
+      - oss/npm_clean_install_with_caching
+      - run: npm run circle
 
 # Important! When adding a new job to `jobs`, make sure to define when it
 # executes by also adding it to the `workflows` section below!
 jobs:
   # Platform tests, each with the same tests but different platform or version.
-  # The docker tag represents the Node.js version and the full list is available
-  # at https://hub.docker.com/r/circleci/node/.
-  Node.js 8:
-    docker: [{ image: "circleci/node:8" }]
-    <<: *common_test_steps
+  NodeJS 8:
+    executor: { name: oss/node, tag: '8' }
+    steps:
+      - common_test_steps
 
-  Node.js 10:
-    docker: [{ image: "circleci/node:10" }]
-    <<: *common_test_steps
+  NodeJS 10:
+    executor: { name: oss/node, tag: '10' }
+    steps:
+      - common_test_steps
+      # We will save the results of this one particular invocation to use in
+      # the publish step. Not only does this make the publishing step take less
+      # time, this also ensures that a passing version gets deployed even if,
+      # theoretically, rebuilding the same commit on the same version of
+      # Node.js should yield the same results!
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./**
 
   VSCode:
-    docker: [{ image: "circleci/node:10" }]
+    executor: { name: oss/node }
     steps:
       - run:
           name: Install deps for VSCode / GUI
           command: sudo apt-get update && sudo apt-get install -y nodejs libgtk3.0 libxss1 libgconf-2-4 libasound2 libnss3 x11-xserver-utils xvfb
-      - *run_install_desired_npm
+      - oss/install_specific_npm_version
       - checkout
-      - restore_cache:
-          keys:
-            # When lock file changes, use increasingly general patterns to restore cache
-            - npm-v2-{{ .Environment.CACHE_VERSION }}-{{ .Branch }}-{{ checksum "package-lock.json" }}
```

Comment:
Note, while I think it's been _ages_ since we've needed to bump the `CACHE_VERSION`, I've moved to a new pattern which leverages a `CIRCLECI_CACHE_VERSION`.  It's still set via the _environment variables_ for the project in CircleCI, and I've already updated it to `v1` (the old `v1` is very much expired by now.)